### PR TITLE
Fix Usability Issues in UI by adding relevant annotations in the Operator

### PR DESCRIPTION
**Pull request by bipuladh**
- Removes Monitoring label, consumed by OLM as the namespace will be labelled automatically from the UI during storage cluster creation. (https://bugzilla.redhat.com/show_bug.cgi?id=1866298)
- Adds description annotation for Storage classes.  (https://bugzilla.redhat.com/show_bug.cgi?id=1888548)

### CODE REVIEW

**Review by umangachapagain — APPROVED** [review]
I'm not sure how adding Description to Annotations is a UI/UX enhancement. But, it doesn't hurt.

**Comment by bipuladh** [comment]
> I'm not sure how adding Description to Annotations is a UI/UX enhancement. But, it doesn't hurt.

These descriptions are picked by the UI and displayed in storage class dropdowns.
Screenshot from 2020-11-03 13-14-49

**Comment by nbalacha** [comment]
The PR title is misleading. These are changes for UX enhancements, not usability issues with the operator. 
Should the storage class annotations highlight that these are Ceph storage classes? The annotations are generic and do not mention the storage is provided by ceph.

**Comment by bipuladh** [comment]
> The PR title is misleading. These are changes for UX enhancements, not usability issues with the operator.
> Should the storage class annotations highlight that these are Ceph storage classes? The annotations are generic and do not mention the storage is provided by ceph.

These are usability issues discovered from the Usability Study. The annotations are created as per the recommendation of the UX team.

**Comment by nbalacha** [comment]
> > The PR title is misleading. These are changes for UX enhancements, not usability issues with the operator.
> > Should the storage class annotations highlight that these are Ceph storage classes? The annotations are generic and do not mention the storage is provided by ceph.
> 
> These are usability issues discovered from the Usability Study. The annotations are created as per the recommendation of the UX team.

These are usability issues in the UX - they are not caused by the operator. This PR adds information to the operator to allow the UI to behave differently. The PR can be reworded to indicate the same.

File: pkg/controller/storagecluster/storageclasses.go [hunk 1]
```diff
@@ -81,6 +81,9 @@ func (r *ReconcileStorageCluster) newOBCStorageClass(initData *ocsv1.StorageClus
 	retSC := &storagev1.StorageClass{
 		ObjectMeta: metav1.ObjectMeta{
 			Name: generateNameForCephRgwSC(initData),
+			Annotations: map[string]string{
+				"description": "Object storage class",
```
Comment (jarrpa) [inline pkg/controller/storagecluster/storageclasses.go:85]:
```suggestion
				"description": "Provides ObjectBucketClaims (OBCs)",
```

File: pkg/controller/storagecluster/storageclasses.go [hunk 2]
```diff
@@ -102,6 +105,9 @@ func (r *ReconcileStorageCluster) newStorageClasses(initData *ocsv1.StorageClust
 		&storagev1.StorageClass{
 			ObjectMeta: metav1.ObjectMeta{
 				Name: generateNameForCephFilesystemSC(initData),
+				Annotations: map[string]string{
+					"description": "File system storage class",
```
Comment (jarrpa) [inline pkg/controller/storagecluster/storageclasses.go:109]:
```suggestion
					"description": "Provides RWO and RWX Filesystem volumes",
```

File: pkg/controller/storagecluster/storageclasses.go [hunk 3]
```diff
@@ -121,6 +127,9 @@ func (r *ReconcileStorageCluster) newStorageClasses(initData *ocsv1.StorageClust
 		&storagev1.StorageClass{
 			ObjectMeta: metav1.ObjectMeta{
 				Name: generateNameForCephBlockPoolSC(initData),
+				Annotations: map[string]string{
+					"description": "Block storage class",
```
Comment (jarrpa) [inline pkg/controller/storagecluster/storageclasses.go:131]:
```suggestion
					"description": "Provides RWO Filesystem volumes, and RWO and RWX Block volumes",
```

**Review by jarrpa — CHANGES_REQUESTED** [review]
I don't like the proposed descriptions. It is meaningless to the user to describe one as a "filesystem" SC and the other as a "block" SC since they can both provide filesystems. The user shouldn't care about the underlying implementation, only what volume and access modes it can provide. I've provided suggestions.

**Comment by bipuladh** [comment]
> I don't like the proposed descriptions. It is meaningless to the user to describe one as a "filesystem" SC and the other as a "block" SC since they can both provide filesystems. The user shouldn't care about the underlying implementation, only what volume and access modes it can provide. I've provided suggestions.

Thanks for the suggestions, I asked the UX folks to have a look at it, and they agree with it, moving forward with your suggestions.

File: pkg/controller/storagecluster/storageclasses.go [hunk 4]
```diff
@@ -14,7 +14,7 @@ import (
 	"k8s.io/apimachinery/pkg/types"
 )
 
-const(
+const (
```
Comment (jarrpa) [inline pkg/controller/storagecluster/storageclasses.go:17]: This change is unrelated and being taken care of in another PR, please remove it.

**Review by nbalacha — APPROVED** [review]

**Review by jarrpa — APPROVED** [review]
/lgtm

**Pull request merged**